x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -92,7 +104,8 @@ synchronized void updatePartialResponse(
         int successfulShards,
         TotalHits totalHits,
         Supplier<InternalAggregations> reducedAggs,
-        int reducePhase
+        int reducePhase,
+        boolean finalReduce /// MP TODO: document me
```

Comment:
This looks odd: updatePartialResponse suggests partial reduction, and it taking a flag `finalReduce` suggests otherwise.

Reply:
That is actually already the case, as AsyncSearchTask calls this method from both
[onPartialReduce](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java#L470) and [onFinalReduce](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java#L484).

It's just that the MutableSearchResponse right now doesn't need to know whether it is the partial or final reduce of the aggs and it doesn't matter for the existing merge logic.

But for the new incremental merge logic for MRT=true, we know need to know in order to determine whether or not to use the partial reduce of local-search aggs. Once the final reduce is true, we no longer want to use the reduction passed in here. We instead want to use the full SearchResponse that is added to the SearchResponseMerge by the CCSActionListener (otherwise we would have double counting/merging of the local results).

If you really don't want this `finalReduce` boolean param here, then we would need to create a new `updateFinalResponse` in MutableSearchResponse that AsyncSearchTask.onFinalReduce would call, but it would do the same thing as `updatePartialResponse` other than setting this new `finalReduce` flag to true.